snews_cs/snews_coinc.py

```diff
@@ -76,140 +61,95 @@ def reset_df(self):
-                continue
-            elif self.cache_df['nu_delta_t'][ind] == 0.0 and not building_new_list:
-                if message['detector_name'] in list(self.cache_df.query(f'sub_list_num=={curr_sub_list_num}')):
-                    continue
-                print(f'Checking this sub list {curr_sub_list_num} ini {self.cache_df["detector_name"][ind]}')
-                if 0 < np.abs(delta_t) <= self.coinc_threshold:
-                    if delta_t < 0:
-                        print('In coincidence with current ini but early')
-                        new_sub_list_num = max_sub_list_num + 1
-                        self.append_message_to_df(message, 0, new_sub_list_num)
-                        self.cache_df = self.copy_row_and_append(ind=ind, delta_t=delta_t,
-                                                                 sub_list_num=new_sub_list_num)
-                        building_new_list = True
-                        continue
-                    else:
-                        print('In coincidence with current list')
-                        self.append_message_to_df(message, delta_t, curr_sub_list_num)
-                        continue
-                elif (np.abs(delta_t) > self.coinc_threshold) and max_sub_list_num == curr_sub_list_num:
-                    print('Building new list: late nu time out of coincidence')
-                    new_sub_list_num = max_sub_list_num + 1
-                    self.append_message_to_df(message, 0, new_sub_list_num)
-                    building_new_list = True
-                    continue
-            if building_new_list and np.abs(delta_t) <= self.coinc_threshold:
-                print('appending to new list')
-                if self.cache_df['detector_name'][ind] in list(self.cache_df.query(f'sub_list_num=={new_sub_list_num}')[
-                                                                   'detector_name']):
-                    print('This detector is already in the list, skipping')
-                    continue
-                else:
-                    print('Adding detector to new sub list.')
-                    self.cache_df = self.copy_row_and_append(ind=ind, delta_t=delta_t,
-                                                             sub_list_num=new_sub_list_num)
-                    continue
+        # after checking all the sublist, if the message not in coincidence with any, only then make a new list
+        if not self.in_coincidence:
+            new_sub_list = max(subs_list_nums, default=-1) + 1
+            self.append_message_to_df(message, 0, new_sub_list)
 
-    # ----------------------------------------------------------------------------------------------------------------
+            # check if any other non-initial signal might be in coincidence
+            other_df = self.cache_df.query(f'sub_list_num!={new_sub_list}')
+            for index, row in other_df.iterrows():
+                self.check_coincidence_melih_seb_edition(row)
```

Comment:
So if nothing was coincident for the message, it makes a new sublist. <br>
Since some messages in other lists may still be in coincidence with this one, I call the same function for the `other_df` and it takes care of everything, because when looping their own sublists they are "already in list" so it only compares every message with this newly created sublist